File: examples/java/src/main/java/examples/copilot/HelloWorld.java
// 01 — Hello world
//
// The smallest useful Copilot SDK program: construct a client, open a session,
// send one prompt, print the answer, shut the client down.
//
// Run: COPILOT_DISABLE_MCP=1 mvn -q exec:java -Dexec.mainClass=examples.copilot.HelloWorld
//
// COPILOT_DISABLE_MCP=1 keeps this example on the bare SDK surface — no
// user-configured MCP servers loaded by the underlying CLI subprocess. The
// JVM cannot mutate its own environment, so the variable must be exported in
// the parent shell. The setProperty mirror below is for any SDK code that
// also consults JVM system properties.
package examples.copilot;

import com.github.copilot.sdk.CopilotClient;
import com.github.copilot.sdk.CopilotSession;
import com.github.copilot.sdk.generated.AssistantMessageEvent;
import com.github.copilot.sdk.json.MessageOptions;
import com.github.copilot.sdk.json.PermissionHandler;
import com.github.copilot.sdk.json.SessionConfig;

public final class HelloWorld {
  public static void main(String[] args) throws Exception {
    System.setProperty("copilot.disable.mcp", "1");
    try (CopilotClient client = new CopilotClient()) {
      client.start().get();

      CopilotSession session = client.createSession(
          new SessionConfig()
              .setModel("gpt-4.1")
              .setOnPermissionRequest(PermissionHandler.APPROVE_ALL)
      ).get();

      AssistantMessageEvent response = session.sendAndWait(
          new MessageOptions().setPrompt("What is 2 + 2?")
      ).get();

      System.out.println(response.getData().content());
    }
  }
}

File: examples/rust/src/bin/hello_world.rs
//! 01 — Hello world
//!
//! The smallest useful Copilot SDK program: open a client, create a session,
//! send one prompt, print the answer, shut the client down.
//!
//! Run: cargo run --bin hello_world

use github_copilot_sdk::{Client, ClientOptions, SessionConfig};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::start(ClientOptions::default()).await?;

    let mut config = SessionConfig::default();
    config.model = Some("gpt-4.1".to_string());
    let session = client.create_session(config).await?;

    if let Some(event) = session.send_and_wait("What is 2 + 2?").await? {
        if let Some(content) = event.data.get("content").and_then(|v| v.as_str()) {
            println!("{}", content);
        }
    }

    session.destroy().await?;
    client.stop().await?;
    Ok(())
}
